common/factoid/fblock.go

```diff
@@ -321,23 +322,34 @@ func (b *FBlock) UnmarshalBinaryData(data []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	cnt, err := buf.PopUInt32()
+	// txLimit is the maximum number of transactions (min 78 bytes) that could
+	// fit into the remaining unread portion of the buffer.
+	txLimit := uint32(buf.Len() / 78)
```

Comment:
Swap ordering of limit and count assignments (?unsure about if this one really matters tho..)